### PR TITLE
Detect duplicate tasks at serialization time

### PR DESCRIPTION
**Pull request by eapolinario**
Signed-off-by: Eduardo Apolinario <eapolinario@users.noreply.github.com>

# TL;DR
Detect if tasks are duplicated at serialization time.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Before we serialize entities in `pyflyte` (both for the `serialize` and `package` commands), we detect if any of the defined tasks is duplicated, and in case they are duplicated, we raise an exception and fail the corresponding `pyflyte` command. We defined two tasks as being duplicated if their metadata identifiers are the same, i.e. if project, domain, name, and version match. 

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1368

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by kumare3** [comment]
Nice 👍🏼

**Review by kumare3 — APPROVED** [review]

File: flytekit/clis/sdk_in_container/serialize.py [hunk 1]
```diff
@@ -123,6 +137,19 @@ def get_registrable_entities(ctx: flyte_context.FlyteContext) -> typing.List:
 
     new_api_model_values = list(new_api_serializable_entities.values())
     entities_to_be_serialized = list(filter(_should_register_with_admin, new_api_model_values))
+    serializable_tasks: typing.List[task_models.TaskSpec] = [
+        entity for entity in entities_to_be_serialized if isinstance(entity, task_models.TaskSpec)
+    ]
+    # Detect if any of the tasks is duplicated. Duplicate tasks are defined as having the same metadata identifiers
+    # (see :py:class:`flytekit.common.core.identifier.Identifier`). Duplicate tasks are considered invalid at registration
```
Comment (wild-endeavor) [inline flytekit/clis/sdk_in_container/serialize.py:144]: There's no need to do RST links in `#` comments - only text in docstrings `""" blah """` get rendered I'm pretty sure.
